issues/4-base-component-called-cask/PLAN.md

```diff
@@ -0,0 +1,166 @@
+# Implementation Plan: Base Component called Cask
+
+## Summary
+
+Create the first iteration of a `Cask` component - a generic container component that serves as a design system primitive. The component allows swapping the underlying HTML tag through an `as` prop while maintaining all features of the target element.
+
+## Scope
+
+### In Scope
+- Create `src/cask/` folder with colocated component and test files
+- Implement `Cask` component with polymorphic `as` prop functionality
+- Set up unit testing framework (Jest + React Testing Library)
+- Add test scripts to package.json and GitHub Actions CI workflow
+- Implement comprehensive unit tests covering:
+  - Default rendering as `div`
+  - Polymorphic rendering with different HTML elements
+  - Props forwarding to underlying elements
+  - TypeScript type safety
+- Update architecture documentation for testing setup and Cask component
+- Ensure all linting, type checking, and tests pass
+
+### Out of Scope
+- Styling/theming system implementation
+- Integration with existing components
+- Performance optimizations beyond basic implementation
+- Advanced polymorphic features (complex element constraints)
+
+## File/Directory Map
+
+```
+src/
+├── cask/
+│   ├── index.ts          # Barrel export
```

Comment:
Please mention that it's not the default export

The component should be usable like so:

```
import {Cask} from '@/src/cask'
```